kubernetes/services/ignite_config.xml

```diff
@@ -43,6 +43,100 @@ Copyright(c) 2020 Futurewei Cloud
+                <property name="name" value="dpm_nodeinfo_cache"/>
+                <!-- Set the cache mode. -->
+                <property name="atomicityMode" value="TRANSACTIONAL"/>
+                <!-- Other cache parameters. -->
+                <property name="cacheMode" value="PARTITIONED"/>
+             </bean>
+
+             <bean class="org.apache.ignite.configuration.CacheConfiguration">
+                <!-- Set the cache name. -->
+                 <property name="name" value="com.futurewei.alcor.dataplane.entity.InternalPorts"/>
+                <!-- Set the cache mode. -->
+                <property name="atomicityMode" value="TRANSACTIONAL"/>
+                <!-- Other cache parameters. -->
+                <property name="cacheMode" value="PARTITIONED"/>
+             </bean>
+
+             <bean class="org.apache.ignite.configuration.CacheConfiguration">
+                <!-- Set the cache name. -->
+                 <property name="name" value="com.futurewei.alcor.dataplane.entity.InternalSubnets"/>
+                <!-- Set the cache mode. -->
+                <property name="atomicityMode" value="TRANSACTIONAL"/>
+                <!-- Other cache parameters. -->
+                <property name="cacheMode" value="PARTITIONED"/>
+             </bean>
+
+             <bean class="org.apache.ignite.configuration.CacheConfiguration">
+                <!-- Set the cache name. -->
+                 <property name="name" value="com.futurewei.alcor.dataplane.entity.InternalSubnetPorts"/>
+                <!-- Set the cache mode. -->
+                <property name="atomicityMode" value="TRANSACTIONAL"/>
+                <!-- Other cache parameters. -->
+                <property name="cacheMode" value="PARTITIONED"/>
+             </bean>
+
+             <bean class="org.apache.ignite.configuration.CacheConfiguration">
+                <!-- Set the cache name. -->
+                 <property name="name" value="com.futurewei.alcor.dataplane.entity.GatewayInfo"/>
+                <!-- Set the cache mode. -->
+                <property name="atomicityMode" value="TRANSACTIONAL"/>
+                <!-- Other cache parameters. -->
+                <property name="cacheMode" value="PARTITIONED"/>
+             </bean>
+
+             <bean class="org.apache.ignite.configuration.CacheConfiguration">
+                <!-- Set the cache name. -->
+                <property name="name" value="ncm_nodeinfo_cache"/>
+                <!-- Set the cache mode. -->
+                <property name="atomicityMode" value="TRANSACTIONAL"/>
+                <!-- Other cache parameters. -->
+                <property name="cacheMode" value="PARTITIONED"/>
+             </bean>
+
+             <bean class="org.apache.ignite.configuration.CacheConfiguration">
+                <!-- Set the cache name. -->
+                 <property name="name" value="com.futurewei.alcor.netwconfigmanager.entity.ResourceMeta"/>
```

Comment:
Is this supposed to be the class path to the cache, or the class path to the cache entity value? 

com.futurewei.alcor.netwconfigmanager.cache.HostResourceMetadataCache?

Reply:
The name used in the config file has to match the canonical name of the class which represents the type of value of the cache. This specific instance, the type/class of the cache is HostResourceMetadataCache (the class containing actual cache data member). The cache data member it self is instantiated as ICache<String, ResourceMeta> hostResourceMeta. This meansresults in:
- Code outside HostResourceMetadataCache will declare a data member of type HostResourceMetadataCache someCache,
- Call memthods exposed by HostResourceMetadataCache  as someCache.put(), someCache.get() etc.,
- Code inside the class it self will call hostResourceMeta.put(), hostResourceMeta.get() etc.,
- Ignite will know this cache by the cononical name of the class of the cache value, which is ResourceMeta, which happens to be com.futurewei.alcor.netwconfigmanager.entity.ResourceMeta.

This explanation of the name Ignite uses to refer to a given cache will be documented in a document on Alcor transactions, caches and services.